docs/public/_redirects

```diff
@@ -1,4 +1,5 @@
 / /x/introduction/ 301
+/x/ https://mui.com/x/ 302
```

Comment:
So we would have https://next.mui.com/x/ as a proxy to https://docs-next--material-ui-x.netlify.app/x/ which would then redirects to https://mui.com/x/. Yeah, it might work. 

If it doesn't, I think that @m4theushw was right in https://github.com/mui/material-ui/pull/34296#discussion_r971947147, we could do:

```diff
diff --git a/docs/public/_redirects b/docs/public/_redirects
index b3ffb080ec..2688d7b27b 100644
--- a/docs/public/_redirects
+++ b/docs/public/_redirects
@@ -405,7 +405,10 @@ https://v4.material-ui.com/* https://v4.mui.com/:splat 301!
 /static/x/* https://docs-next--material-ui-x.netlify.app/static/x/:splat 200
 /static/* https://mui.com/static/:splat 200
 /x/_next/* https://docs-next--material-ui-x.netlify.app/_next/:splat 200
-/x/* https://docs-next--material-ui-x.netlify.app/x/:splat 200
+/x/introduction/* https://docs-next--material-ui-x.netlify.app/x/introduction/:splat 200
+/x/react-data-grid/* https://docs-next--material-ui-x.netlify.app/x/react-data-grid/:splat 200
+/x/react-date-pickers/* https://docs-next--material-ui-x.netlify.app/x/react-date-pickers/:splat 200
+/x/api/* https://docs-next--material-ui-x.netlify.app/x/api/:splat 200
 /r/x-* https://docs-next--material-ui-x.netlify.app/r/x-:splat 200
 /:lang/x/* https://docs-next--material-ui-x.netlify.app/:lang/x/:splat 200
 ```
 
 So that https://next.mui.com/x/ redirects to https://mui.com/x/.

Reply:
> but https://next.mui.com/x/introduction/ still works and is a duplicate of https://mui.com/x/introduction/.

@m4theushw What do you mean by it's a duplicate? This link points to the right branch on each URL:

<img width="145" alt="Screenshot 2022-10-26 at 13 41 32" src="https://user-images.githubusercontent.com/3165635/198017228-1a19ebaa-4d7b-471c-be46-16d10a60d5cf.png">